### ISSUE TITLE
Error saving to to Safetensor format: "ERROR is You are trying to save a non contiguous tensor: `model.diffusion_model.input_blocks.0.0.weight` which is not allowed. It either means you are trying to save tensors which are reference of each other in which case it's recommended to save only the full tensors, and reslice at load time, or simply call `.contiguous()` on your tensor to pack it before saving.

### ISSUE THREAD

**Issue by CCpt5** [wkpark/sd-webui-model-mixer#52]
TL;DR: Been getting error saving models out to Safetensors format for a while - always worked around saving to CKPT. Testing new LoRA export realized I will need safetensor functionality working to use that upcoming feature.

-----
Hello again!

I get the following error whenever I try to save out a merge to .safetensors format:

"ERROR is You are trying to save a non contiguous tensor: `model.diffusion_model.input_blocks.0.0.weight` which is not allowed. It either means you are trying to save tensors which are reference of each other in which case it's recommended to save only the full tensors, and reslice at load time, or simply call `.contiguous()` on your tensor to pack it before saving."

I've been getting this error for a little while, but I've just untoggled the safetensors option and saved to ckpt instead and that's worked without issue. Howver, I pulled the development branch w/ early LoRA export functionality and I realized this above error might cause issues when exporting to LoRA. 

My first attempts exporting to Lora and Lycoris worked great! But - for some reason the exported LoRAs don't show up in A1111 unless I turn off this option in settings, "Always show all networks on the Lora page (otherwise, those detected as for incompatible version of Stable Diffusion will be hidden)"

So I'm thinking A1111s not recognizing the Model Mixer generated LoRAs as valid for SDXL - My guess is that it's related to the safetensor error I get when trying to save full models out to Safetensors. I don't have a workaround that would work in this case.
  
Really sorry if this is user error (Re: something in my A111 settings), but I do get it on both the current 1.6 version and the latest development version v1.6.0-295-g9c1c0da0 which has the tweaks Automatic added recently related to the other issue in this forum.

Thanks so much for everything - love the extension!

**Comment by wkpark** [comment]
> TL;DR: Been getting error saving models out to Safetensors format for a while - always worked around saving to CKPT. Testing new LoRA export realized I will need safetensor functionality working to use that upcoming feature.
> 
> Hello again!
> 
> I get the following error whenever I try to save out a merge to .safetensors format:
> 
> "ERROR is You are trying to save a non contiguous tensor: `model.diffusion_model.input_blocks.0.0.weight` which is not allowed. It either means you are trying to save tensors which are reference of each other in which case it's recommended to save only the full tensors, and reslice at load time, or simply call `.contiguous()` on your tensor to pack it before saving."
> 

full error log needed to track easily

> I've been getting this error for a little while, but I've just untoggled the safetensors option and saved to ckpt instead and that's worked without issue. Howver, I pulled the development branch w/ early LoRA export functionality and I realized this above error might cause issues when exporting to LoRA.
> 
> My first attempts exporting to Lora and Lycoris worked great! But - for some reason the exported LoRAs don't show up in A1111 unless I turn off this option in settings, "Always show all networks on the Lora page (otherwise, those detected as for incompatible version of Stable Diffusion will be hidden)"
> 

I guess, the metadata exported by model-mixer is not good enough.

> So I'm thinking A1111s not recognizing the Model Mixer generated LoRAs as valid for SDXL - My guess is that it's related to the safetensor error I get when trying to save full models out to Safetensors. I don't have a workaround that would work in this case.
> 
> Really sorry if this is user error (Re: something in my A111 settings), but I do get it on both the current 1.6 version and the latest development version v1.6.0-295-g9c1c0da0 which has the tweaks Automatic added recently related to the other issue in this forum.
> 
> Thanks so much for everything - love the extension!

Thank you for your feed back so quickly!

**Comment by wkpark** [comment]
* [x] SDXL LoRA/LyCORIS needs `ss_base_model_version` metadata attribute as `sdxl_base_v1-0` - fixed
* [x] fix for `contiguous` check in the `safetensors.torch.save_file()` - fixed
